src/SLCore.IntegrationTests/DependencyLocator.cs

```diff
@@ -0,0 +1,90 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.Linq;
+using System.Xml;
+
+namespace SonarLint.VisualStudio.SLCore.IntegrationTests;
+
+public class DependencyLocator // this might be reused in the product code in the future
+{
+    public static List<string> AnalyzerPlugins { get; private set; }
+    public static string Sloop { get; private set; }
+
+    static DependencyLocator()
+    {
+        var localAppData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+        
+        var dependencyProps = new XmlDocument();
+        dependencyProps.Load("EmbeddedSonarAnalyzer.props");
+        
+        EnsureSloopIsAvailable(localAppData, dependencyProps);
+        EnsurePluginsAreAvailable(localAppData, dependencyProps);
+    }
+
+    private static void EnsurePluginsAreAvailable(string localAppData, XmlDocument dependencyProps)
+    {
+        var jarDirectoryPath = Path.Combine(localAppData, "SLVS_Build_DownloadedJars");
+        var availablePluginJars = Directory.GetFiles(jarDirectoryPath);
+        var roslynAnalyzerVersion = GetDependencyVersion("EmbeddedSonarAnalyzerVersion", dependencyProps);
+        var cfamilyAnalyzerVersion = GetDependencyVersion("EmbeddedSonarCFamilyAnalyzerVersion", dependencyProps);
+        var jstsAnalyzerVersion = GetDependencyVersion("EmbeddedSonarJSAnalyzerVersion", dependencyProps);
+        var secrestsAnalyzerVersion = GetDependencyVersion("EmbeddedSonarSecretsJarVersion", dependencyProps);
+        AnalyzerPlugins = new List<string>
+        {
+            GetAnalyzerPath("sonar-csharp-plugin", roslynAnalyzerVersion, availablePluginJars),
+            GetAnalyzerPath("sonar-vbnet-plugin", roslynAnalyzerVersion, availablePluginJars),    
+            GetAnalyzerPath("sonar-cfamily-plugin", cfamilyAnalyzerVersion, availablePluginJars),
+            GetAnalyzerPath("sonar-javascript-plugin", jstsAnalyzerVersion, availablePluginJars),
+            GetAnalyzerPath("sonar-text-plugin", secrestsAnalyzerVersion, availablePluginJars),
+        };
+    }
+
+    private static void EnsureSloopIsAvailable(string localAppData, XmlDocument dependencyProps)
+    {
+        var sloopVersion = GetDependencyVersion("EmbeddedSloopVersion", dependencyProps);
+        var sloopPath = Path.Combine(localAppData,
+            "SLVS_Build_Sloop",
+            $"sonarlint-backend-cli-{sloopVersion}-windows_x64",
+            "bin",
+            "sonarlint-backend.bat");
+        if (!File.Exists(sloopPath))
+        {
+            throw new InvalidOperationException();
```

Comment:
Add a message to the exception so we can see what the problem is without debugging.